xgi/algorithms/centrality.py

```diff
@@ -249,3 +250,59 @@ def node_edge_centrality(
     return {node_dict[n]: new_x[n] for n in node_dict}, {
         edge_dict[e]: new_y[e] for e in edge_dict
     }
+
+
+def line_vector_centrality(H):
+    """The vector centrality of nodes in the line graph of the hypergraph.
+    Parameters
+    ----------
+    H : Hypergraph
+        The hypergraph of interest
+
+    Returns
+    -------
+    dict
+        Centrality, where keys are node IDs and values are lists of centralities.
+
+    References
+    ----------
+    Vector centrality in hypergraphs,
+    K. Kovalenko, M. Romance, E. Vasilyeva, D. Aleja, R. Criado, D. Musatov,
+    A.M. Raigorodskii, J. Flores, I. Samoylenko, K. Alfaro-Bittner, M. Perc, S. Boccaletti,
+    https://doi.org/10.1016/j.chaos.2022.112397
+    """
+
+    if not xgi.is_connected(H):
+        raise XGIError("This method is not defined non-connected hypergraphs.")
+
+    LG = convert_to_line_graph(H)
+    LGcent = nx.eigenvector_centrality(LG)
+
+    vc = {node: [] for node in H.nodes}
+
+    edge_label_dict = {
+        tuple(edge): index for index, edge in enumerate(H.edges.members())
+    }
+    hyperedge_dims = {tuple(edge): len(edge) for edge in H.edges.members()}
+
+    D = H.edges.size.max()
+
+    for k in range(2, D + 1):
+        c_i = np.zeros(len(H.nodes))
+
+        for edge, _ in list(filter(lambda x: x[1] == k, hyperedge_dims.items())):
```

Comment:
Should also work with (check if not slower). Then you don't need to create `hyperedge_dims` anymore 
```suggestion
        for edge in H.edges.filterby("size", k).members():
```